queue.tf

```diff
@@ -0,0 +1,67 @@
+# ------------------------------------------------------------------------------
+# Provision the CloudTrail SQS queue that will collect notifications
+# of logs being written to the CDM CloudTrail.
+# ------------------------------------------------------------------------------
+
+# CDM queue of messages notifying of CloudTrail logs being written to
+# the CDM S3 bucket
+resource "aws_sqs_queue" "cloudtrail" {
+  # We can't perform this action until our policy is in place.
+  depends_on = [
+    aws_iam_role_policy_attachment.provisioncdmcloudtrail_policy_attachment,
+  ]
+
+  name = var.queue_name
+  redrive_policy = jsonencode({
+    deadLetterTargetArn = aws_sqs_queue.cloudtrail_dlq.arn
+    # Move messages to the DLQ if the user can't process them after
+    # five tries.
+    maxReceiveCount = 4
```

Comment:
Just confirming that `maxReceiveCount = 4` really means the `five tries` specified in the comment and that this isn't a typo?

Reply:
From [the AWS docs](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html) (emphasis mine):
> `maxReceiveCount` – The number of times a message is delivered to the source queue before being moved to the dead-letter queue. When the `ReceiveCount` for a message _exceeds) the `maxReceiveCount` for a queue, Amazon SQS moves the message to the dead-letter-queue.